src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java

```diff
@@ -69,6 +69,12 @@ public boolean canHandle(
 
         if (callback.getEvent() == Event.START_APPEAL) {
             asylumCase.write(APPEAL_REFERENCE_NUMBER, DRAFT);
+
+            String caseId = String.valueOf(callback.getCaseDetails().getId());
+            caseId = caseId.replaceAll("(.{" + 4 + "})", "$1 ").trim();
+
+            asylumCase.write(CCD_REFERENCE_NUMBER_FOR_DISPLAY,  caseId);
+
```

Comment:
@Lee could you look to move this code into a new handler (i.e. a handler that will be triggered for all events) so the CCD_REFERENCE_NUMBER_FOR_DISPLAY is the available for all appeals.

If you take a look at the `AsylumCaseDataFixingHandler` class, you will see that the `canHandle()` method always returns true - so a handler similar to this. 

If we have the code in the `AppealReferenceNumberHandler`, the CCD_REFERENCE_NUMBER_FOR_DISPLAY will only display/be written to the AsylumCase for new cases.

Reply:
Yes, I agree. I think the cleanest way is to use new dedicated handler for it.
There is an option to use fixers but the whole part of the fixers should be refactored before they are more useful.

Reply:
Changed to use new handler - AsylumCaseCcdReferenceNumberHandler